### PR TITLE
Split godoc into separate packages

### LINKED ISSUE #305: docs do not explain how to paginate
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/go-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://pkg.go.dev/github.com/auth0/go-auth0) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/go-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/go-auth0/releases/latest) of this SDK and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

The documentation does not explain how to use pagination.

PS: bonus bug: the link in the PR template to the Auth0 Community forums (`https://community.auth0.com/c/sdks/5`) 404s

### Expectation

A concise explanation of pagination using the go-auth0 sdk, including
* a code snippet showing how to do it
* a discussion of whether sorting is needed or not
* explanation of tradeoffs using `List.Next` vs page indexes
* links from each method returning a `List` to the explanation

### Reproduction

Infinite loop doing the naive thing:
```
auth0API, _ := management.New(...)
ul, _ := auth0API.User.List(ctx)
for {
	for _, u := range ul.Users {
		// process user
	}

	if !ul.HasNext() {
		break
	}
	ul, _ = auth0API.User.List(ctx, management.From(ul.Next))
}
```

### Auth0 Go SDK version

v1.2.0

### PR DESCRIPTION
**Pull request by ewanharris**
### 🔧 Changes

This PR splits the godoc into the separate packages rather than having both the auth and management client documentation in the top level.

Some screenshots of the docs below

<details>
<summary>Auth0 package page</summary>

Screenshot 2023-11-10 at 10 46 27

</details>

<details>
<summary>Authentication package page</summary>

Screenshot 2023-11-10 at 10 46 39

</details>

<details>
<summary>Manangement package page</summary>

Screenshot 2023-11-10 at 10 49 44

</details>

If you want to view this locally then you can install godoc (`go install  golang.org/x/tools/cmd/godoc@latest`) and then run `godoc -http=:6060` in the repo. The documentation will be available on `http://localhost:6060/pkg/github.com/auth0/go-auth0/`

### 📚 References

Fixes #305 alongside the refactor

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [ ] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -1,129 +1,75 @@
...
+//	c := &management.Client{
+//		Name:        auth0.String("Client Name"),
+//		Description: auth0.String("Long description of client"),
+//	}
+//	err = m.Client.Create(context.Background(), c)
+//	if err != nil {
+//		// handle err
+//	}
+//
+// [management client docs]: https://pkg.go.dev/github.com/auth0/go-auth0/management
```
Comment (ewanharris) [inline doc.go:71]: I'm not super thrilled about having to link to pkg.go.dev directly but I'm not sure the [doc links](https://go.dev/doc/comment#links:~:text=in%20HTML%20renderings.-,Doc%20links,-Doc%20links%20are) support linking like I want to here (i.e to the top level docs)

File: authentication/doc.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// # Usage
+//
+//	import (
+//		"github.com/auth0/go-auth0"
+//		"github.com/auth0/go-auth0/authentication"
+//		"github.com/auth0/go-auth0/authentication/database"
+//		"github.com/auth0/go-auth0/authentication/oauth"
+//	)
+//	a, err := authentication.New(
+//		context.Background(),
```
Comment (Widcket) [inline authentication/doc.go:12]: Should we use `context.TODO()` (https://github.com/auth0/go-auth0/pull/307)?
Also for other instances in the file.

File: doc.go [hunk 3]
```diff
@@ -1,129 +1,75 @@
...
+//
+//	go get github.com/auth0/go-auth0
+//
+// # Authentication
+//
+// Below is an example of using the Authentication client, for full documentation visit the
+// [authentication client docs].
+//
+//	authAPI, err := authentication.New(
+//		context.Background(),
```
Comment (Widcket) [inline doc.go:24]: Same as above, and for other instances in this file.

File: doc.go (hunk 1 above)
Comment (Widcket) [reply doc.go:71]: Would a relative link work?

Screenshot 2023-11-10 at 13 36 29

File: authentication/doc.go (hunk 2 above)
Comment (ewanharris) [reply authentication/doc.go:12]: Ah yeah I totally forgot to update the code here when making that change 🙃 Thanks for catching!

File: management/doc.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+// A management client can be instantiated a domain, client ID and secret.
+//
+//	import (
+//		"github.com/auth0/go-auth0"
+//		"github.com/auth0/go-auth0/management"
+//	)
+//	// Initialize a new client using a domain, client ID and secret.
+//	m, err := management.New(
+//		domain,
+//		management.WithClientCredentials(context.Background(), id, secret),
```
Comment (Widcket) [inline management/doc.go:14]: Same as above (`context.TODO()`), and for other instances in this file.

File: management/doc.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
+// Package management provides a client for using the Auth0 Management API
```
Comment (Widcket) [inline management/doc.go:1]:
```suggestion
// Package management provides a client for using the Auth0 Management API.
```

File: doc.go (hunk 1 above)
Comment (ewanharris) [reply doc.go:71]: Unfortunately not, we can't use HTML in the docs and the URL support only seems to support full URLs rather than relative.

**Review by Widcket — APPROVED** [review]

**Pull request merged**